src/freenet/node/PeerLocation.java

```diff
@@ -1,14 +1,16 @@
 package freenet.node;
 
 import java.util.Arrays;
+import java.util.Set;
 
 import freenet.support.Logger;
 
 public class PeerLocation {
 	
 	/** Current location in the keyspace, or -1 if it is unknown */
 	private double currentLocation;
-	/** Current locations of our peer's peers */
+	/** Current sorted array of locations of our peer's peers. Must not be modified,
+	 may only be replaced entirely. */
 	private double[] currentPeersLocation;
```

Comment:
Just to clarify, what I'm suggesting is to factor https://github.com/bertm/fred-staging/blob/ca8ff8a171d187b0e4678b15950fae8f9482d44c/src/freenet/node/PeerManager.java#L1047 inside the same for loop